<launch>
    <node pkg="mobile_robot_comms" type="static_tf_publisher_node" name="static_tf_publisher">
    </node>

    <node name="rplidarNode_front" pkg="rplidar_ros"  type="rplidarNode" output="screen">
	<remap from="scan" to="scan_front" />
	<param name="serial_port"         type="string" value="/dev/ttyUSB1"/>
	<param name="serial_baudrate"     type="int"    value="256000"/>
	<param name="frame_id"            type="string" value="laser_front"/>
	<param name="inverted"            type="bool"   value="false"/>
	<param name="angle_compensate"    type="bool"   value="true"/>
    </node>

    <node name="rplidarNode_rear" pkg="rplidar_ros"  type="rplidarNode" output="screen">
	<remap from="scan" to="scan_rear" />
	<param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
	<param name="serial_baudrate"     type="int"    value="256000"/>
	<param name="frame_id"            type="string" value="laser_rear"/>
	<param name="inverted"            type="bool"   value="false"/>
	<param name="angle_compensate"    type="bool"   value="true"/>
    </node>

    <node pkg="mobile_robot_comms" type="base_controller_node" name="base_controller">
        <param name="axis_length" value="0.33" />
        <param name="wheel_radius" value="0.09" />
        <param name="ticks_rev" value="520" />
        <param name="max_linear_velocity" value="0.45" />
        <param name="max_angular_velocity" value="1.0" />
    </node>
    <node pkg="mobile_robot_comms" type="odometry_publisher_node" name="odometry_publisher"> 
        <param name="axis_length" value="0.33" />
        <param name="wheel_radius" value="0.09" />
        <param name="ticks_rev" value="520" />
        <param name="lwheel_alpha" value="0.5" />
        <param name="rwheel_alpha" value="0.5" />
    </node>

    <node pkg="mobile_robot_comms" type="laser_filter_node" name="laser_filter">
        <param name="min_range" value="0.27" />
    </node>
</launch>
